### DISCUSSION TITLE
Creating Rounded Rectangle and Ring on the Screen

### DISCUSSION THREAD

**Discussion by nebsar** [pelicanmapping/osgearth#1964] [Support Q&A] [answered]
I am trying to add rounded rectangle on the screen with the following code. When I set the buffer value to the half of the height and with which is 400, rounded rectangle becomes a Ring. This is good, But the Ring is not smooth enough. When, looked carefully, it looks like a polygon with lots of vertices. As I know, rings are drawn as polygon vertices. My question is; how can I make the Ring or Rounded Rectangle's edges smoother? Is this the best way to add a geometry on the screen?

float buffer = Geometry::hasBufferOperation() ? 200.0f : 0.0f;

	osg::ref_ptr<Geometry> geome = new Polygon();
	
	geome->push_back(osg::Vec3d(buffer, buffer, 0));
	geome->push_back(osg::Vec3d(400- 1 - buffer, buffer, 0));
	geome->push_back(osg::Vec3d(400 - 1 - buffer, 400 - 1 - buffer, 0));
	geome->push_back(osg::Vec3d(buffer, 400 - 1 - buffer, 0));

	BufferParameters bp;
	bp._capStyle = BufferParameters::CAP_ROUND;
	geome->buffer(buffer - 1.0f, geome, bp);

	GeometryRasterizer ras((int)400, (int)400);
	ras.draw(geome.get(), osg::Vec4f(0,0,0,0.4));

	osg::Image* image = ras.finalize();
	ImageControl* s_imageControl = new ImageControl(image);
	right->addControl(s_imageControl);

![image](https://user-images.githubusercontent.com/5169314/152185984-8cadd7ed-4746-4e07-94c7-bae59e7e60ed.png)

**Comment by gwaldron** [comment, marked as answer]
The buffering is a GEOS operation so you may wish to consult the GEOS code for an answer. Have you tried the `BufferParameters._cornerSegs` property?

**Reply by nebsar** [reply]
BufferParameters._cornerSegs worked. Thank you Glenn.
